### DISCUSSION TITLE
Help on models.py

### DISCUSSION THREAD

**Discussion by PARKER-X** [GeriLife/companionship-care#201] [Q&A] [answered]
@property
    def annotated_companions(self):
        """
        Return a companion list annotated with activity count for current persion.
        TODO: refactor for performance,
            such as by adding a cached "activity_count" property to Companion model
        """
        annotated_companions = []

        for member in self.companions_through.all():
            member.activity_count = member.get_activity_count(circle=self)

            annotated_companions.append(member)

        return annotated_companions

class  Companion(models.Model):

    def get_activity_count(self, circle=None):
        """Return a count of activities between the related user and circle."""
        return self.user.get_activity_count(circle=self.circle)

How User gets attribute get_activity_count here??

**Comment by brylie** [comment, marked as answer]
The `User` model defines the `get_activity_count` method.

https://github.com/CompanionshipCare/companionship-care/blob/66f33b7926feaac897513db2407eae006c181a1f/project/accounts/models.py#L68
